src/syft/lib/numpy/array.py

```diff
@@ -0,0 +1,39 @@
+# third party
+import numpy as np
+import torch
+
+# syft relative
+from ...generate_wrapper import GenerateWrapper
+from ...lib.torch.tensor_util import protobuf_tensor_deserializer
+from ...lib.torch.tensor_util import protobuf_tensor_serializer
+from ...proto.lib.torch.tensor_pb2 import TensorData
+
+
+def object2proto(obj: np.ndarray) -> TensorData:
+    tensor = torch.from_numpy(obj).clone()
+    tensor_proto = protobuf_tensor_serializer(tensor)
+    # TODO:
+    # support numpy.object_, numpy.str_, numpy.unicode_,
+    #         numpy.uint16, numpy.uint32, numpy.uint64,
```

Comment:
If we add a custom Numpy proto which contains an additional field which is the numpy type as a string.

Then we should be able to do something like this:
```python
import numpy as np
import torch

# our unsupported numpy array
x = np.array([0, 1, 2], dtype=np.uint16)
npdtype = x.dtype

# refactor to a dict of uint -> int types for matching
if npdtype == np.uint16:
    x = x.astype(np.int16)

y = torch.from_numpy(x)

# serialize along side the original dtype
ser_y = y
ser_numpy_dtype = npdtype.name


# deserialize on other side
y = ser_y
numpy_dtype = ser_numpy_dtype

# conver to numpy first
z = y.numpy()

# get numpy type from string
npdtype = np.dtype(numpy_dtype)

# down sample
z = z.astype(npdtype)

assert z.all() == x.all()
assert z.dtype == np.uint16
```

Leave off complex, unicode, str and object for now and just throw exceptions and we can make it seperate tasks to handle them as a proper Numpy Array serde protobuf.

Reply:
Please correct me if I'm wrong, could I also ignore uint and raise exceptions or should I make a custom proto which supports uint now?

Reply:
I would add it now since the majority of things we want numpy for is numbers so it would be good if we dont hit any of these by accident and have it not work.

To add the custom proto just create one that looks something like this:
```proto
syntax = "proto3";

package syft.lib.numpy;

import "proto/lib/torch/tensor.proto";

message NumpyProto {
  syft.lib.torch.TensorData tensor = 1;
  string dtype = 2;
}

```

Then run: `scripts/build_proto.sh`.

Reply:
Thank you for your clear instruction! I implemented this, and it worked! We can now use uint!